src/modules/workspaces/workspaces.usecase.ts

```diff
@@ -450,6 +450,90 @@ export class WorkspacesUsecases {
     return { result };
   }
 
+  async getUserUsageInWorkspace(
+    user: User,
+    workspaceId: WorkspaceAttributes['id'],
+  ) {
+    const member = await this.workspaceRepository.findWorkspaceUser({
+      memberId: user.uuid,
+      workspaceId,
+    });
+
+    if (!member) {
+      throw new BadRequestException('User not valid');
+    }
+
+    const syncStartedAt = new Date();
```

Comment:
Get the date from here to prevent cases such as some file being set as DELETED while this is running. 

This could happen if the user deletes a very nested folder and children files are still being updated by the trigger in the database. 

The next time the sync is ran, it should include the files that were not calculated in this call.